### PR TITLE
Aggregate per process result

### PR DESCRIPTION
**Pull request by LuQQiu**
Aggregate per process result to show to user.

Example usage
uv run bench.py -t 2 -q 100 -p lu_prefix
uv run multi.py -n 2 -p lu_prefix -q 100 -t 2 
uv run multi.py -n 2 -p lu_prefix -q 100 -t 2 --no-ingest --no-index -qn 2

### CODE REVIEW

File: multi.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        "prefix": prefix,  # Base prefix, will be modified per process
+        "reset": reset,
+    }
+
+    process_args = []
+
+    if ingest or index:
+        for i in range(0, num_processes):
+            process_kwargs = bench_kwargs.copy()
+            process_args.append((i + 1, 0, process_kwargs))
```
Comment (rpgreen) [inline multi.py:62]: This is hard to follow

File: multi.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        "-n",
+        "--num_processes",
+        type=int,
+        required=False,
+        default=1,
+        help="Number of total benchmark process. This number should be the same for data ingestion and data querying.",
+    )
+    parser.add_argument(
+        "-qn",
+        "--query_process",
```
Comment (rpgreen) [inline multi.py:106]: Why separate parameters for querying and ingestion? This might be confusing for users

**Review by rpgreen — COMMENTED** [review]
Please make sure to update the readme

**Comment by LuQQiu** [comment]
Close to leave this branch unchanged for user. Create a new PR a new branch, will keep the review comments

File: multi.py (hunk 2 above)
Comment (LuQQiu) [reply multi.py:106]: This is because our previous script is optimized for total QPS. Our total QPS across all tables are very good, but per table QPS is around 18 which harvey is asking for 50 or even more.

having multiple process querying a table can probably give us best per table QPS perf.

Any suggestion for that? should i change to use multiple threads querying a table? but threads in python is not actually multi-threading. Haven't done the tests, not quite sure about the overall perf of this other approach

```
Single table concurrent read test
uv run multi.py -n 1 -t 1 -p table_prefix -q 100 -r
create 1 process, each create 1 tables with 1 million rows with table name with table_prefix
for ingestion and indexing
and then run
uv run multi.py -n 1 -t 1 -p table_prefix -q 10000 -qn 10 -no-ingest --no-index
to run 10 processes (-qn query process number) to each run 10000 queries against the created single table (edited)
```

**Comment by LuQQiu** [comment]
Please refer to https://github.com/lancedb/lancedb-cloud-benchmarks/pull/8
